src/Microsoft.Extensions.OData.Migration/Formatters/Serialization/ODataMigrationOutputFormatter.cs

```diff
@@ -0,0 +1,385 @@
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static string GetRootElementName(ODataPath path)
+        {
+            if (path != null)
+            {
+                ODataPathSegment lastSegment = path.Segments.LastOrDefault();
+                if (lastSegment != null)
+                {
+                    OperationSegment actionSegment = lastSegment as OperationSegment;
+
+                    if (actionSegment != null)
+                    {
+                        IEdmAction action = actionSegment.Operations.Single() as IEdmAction;
+                        if (action != null)
+                        {
+                            return action.Name;
+                        }
+                    }
+
+                    PropertySegment propertyAccessSegment = lastSegment as PropertySegment;
+                    if (propertyAccessSegment != null)
+                    {
+                        return propertyAccessSegment.Property.Name;
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        // Return what is preferred (not very relevant in v3)
+        private static string GetRequestPreferHeader(IHeaderDictionary headers)
+        {
+            string PreferHeaderName = "Prefer";
+            StringValues stringValues;
+            bool found = headers.TryGetValue(PreferHeaderName, out stringValues);
+            IEnumerable<string> values = stringValues.AsEnumerable();
+
+            if (found)
+            {
+                // If there are many "Prefer" headers, pick up the first one.
+                return values.FirstOrDefault();
+            }
+
+            return null;
+        }
+
+        // Analyzes incoming type, then uses Serializer Provider through functions to return appropriate serializer
+        private static ODataSerializer GetSerializer(Type type, object value, HttpRequest internalRequest, Func<IEdmTypeReference, ODataSerializer> getEdmTypeSerializer, Func<Type, ODataSerializer> getODataPayloadSerializer)
+        {
+            ODataSerializer serializer;
+
+            IEdmObject edmObject = value as IEdmObject;
+            if (edmObject != null)
+            {
+                IEdmTypeReference edmType = edmObject.GetEdmType();
+                if (edmType == null)
+                {
+                    throw new SerializationException("Edm type " + edmObject.GetType().FullName + " cannot be null");
+                }
+
+                serializer = getEdmTypeSerializer(edmType);
+                if (serializer == null)
+                {
+                    string message = "Type " + edmType.ToTraceString() + " cannot be serialized";
+                    throw new SerializationException(message);
+                }
+            }
+            else
+            {
+                var applyClause = internalRequest.ODataFeature()?.ApplyClause;
```

Comment:
Don't use var unless the actual type is obvious from just reading the code (i.e. not using an IDE).  That's not the case here.